build/scripts/entrypoint-init-container.sh

```
@@ -70,7 +70,12 @@ else
     echo "Using editor download: $ide_download_url"
 fi
 
-curl -fsSL "$ide_download_url" | tar xzf - --strip-components=1
+# Bypass download + extraction if already done (for same binary)
+prev_ide_download=$([ -e .ide-download-complete ] && cat .ide-download-complete || echo "")
+if [ "${prev_ide_download}" != "${ide_download_url}" ]; then
+  curl -fsSL "$ide_download_url" | tar xzf - --strip-components=1
+  echo -n "${ide_download_url}" > .ide-download-complete
```

Shellcheck (0.11.0)

[warning] In POSIX sh, echo flags are undefined.

(SC3037)